homework/index.js

```diff
@@ -1,53 +1,143 @@
-    createAndAppend('li', ul, { text: repo.name });
+    const listItem = createAndAppend('li', ul, { class: 'listItem' });
+    const table = createAndAppend('table', listItem, { class: 'table' });
+    const topRow = createTable(table, 'Repository: ');
+    createAndAppend('a', topRow.lastChild, {
+      text: repo.name,
+      href: repo.html_url,
+    });
+    createTable(table, 'Description: ', repo.description);
+    createTable(table, 'Fork count: ', repo.forks);
+    createTable(table, 'Created on: ', changeDateFormat(repo.created_at));
+    createTable(table, 'Updated on: ', changeDateFormat(repo.updated_at));
+  }
+
+  function contributorsSection(url, parent) {
+    fetchJSON(url)
+      .then(repos => {
+        repos.forEach(repo => {
+          const li = createAndAppend('li', parent, { class: 'rightList' });
+          createAndAppend('img', li, {
+            class: 'image',
+            src: repo.avatar_url,
+          });
+          createAndAppend('a', li, {
+            text: repo.login,
+            href: repo.html_url,
+            target: '_blank',
+            class: 'cont-link',
+          });
+          createAndAppend('p', li, {
+            class: 'contributions',
+            text: repo.contributions,
+          });
+        });
+      })
+      .catch(err => {
+        createAndAppend('div', parent, {
+          text: err.message,
+          class: 'alert-error',
+        });
+      });
+  }
+
+  function repoSelect(item, repoItem, parent) {
+    const ul = createAndAppend('ul', parent);
+    const repo = repoItem[item];
+    renderRepoDetails(repo, ul);
   }
 
   function main(url) {
-    fetchJSON(url, (err, repos) => {
-      const root = document.getElementById('root');
-      if (err) {
+    const root = document.getElementById('root');
+    const header = createAndAppend('header', root, {
+      text: 'HYF Repositories',
+    });
+    const mainElem = createAndAppend('main', root, { class: 'main-container' });
+    const repoContainer = createAndAppend('section', mainElem, {
+      class: 'repo-container',
+    });
+    const ul = createAndAppend('ul', repoContainer);
+    const rightSection = createAndAppend('section', mainElem, {
+      class: 'contributors-container',
+    });
+    const p = createAndAppend('p', rightSection, { class: 'right-title' });
+    p.innerHTML = 'Contributions';
+    const rightUl = createAndAppend('ul', rightSection, { class: 'right-ul' });
+    const select = createAndAppend('select', header, { class: 'test' });
+    fetchJSON(url)
+      .then(repoItem => {
+        renderRepoDetails(repoItem[17], ul);
+        const startItem = repoItem[17].contributors_url;
+        contributorsSection(startItem, rightUl);
+        repoItem
+          .sort((a, b) => a.name.localeCompare(b.name))
+          .forEach((repo, index) => {
+            createAndAppend('option', select, {
+              text: repo.name,
+              value: index,
+            });
+          });
```

Comment:
As I mentioned in the feedback of @ahmthsrvsmz with whom you teamed up on this, the solution to rendering the first repo is simple: first sort, then render:

```js
fetchJSON(url)
  .then(repos => {
    repos
      .sort((a, b) => a.name.localeCompare(b.name))
      .forEach((repo, index) => {
        createAndAppend('option', select, {
          text: repo.name,
          value: index,
        });
      });
    renderRepoDetails(repos[0], ul);
    const startItem = repos[0].contributors_url;
    contributorsSection(startItem, rightUl);
```

Reply:
First of all, I should indicate it better Jim. I have only got help from  @ahmthsrvsmz for a small part of homework, such as listing only one repo on page load. Sorry for possible misleading if it's the case.
And the feedback for rendering, it's been piece of cake, thanks for the insight!

Reply:
Are there any updates that I can review? This PR is still not approved.

Reply:
Hi Jim, indeed there are but still not ready to push. I really wished that I've submitted a renewed homework at least today. I'm planning to do so tomorrow afternoon. Thank you for your message!